Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/WiimoteProfileStringSetting.java

```diff
@@ -0,0 +1,67 @@
+package org.dolphinemu.dolphinemu.features.settings.model;
+
+import org.dolphinemu.dolphinemu.features.settings.utils.SettingsFile;
+
+// This stuff is pretty ugly. It's a kind of workaround for certain controller settings
+// not actually being available as game-specific settings.
+public class WiimoteProfileStringSetting implements AbstractStringSetting
+{
+  private final int mPadID;
+
+  private final String mSection;
+  private final String mKey;
+  private final String mDefaultValue;
+
+  private final String mProfileKey;
+
+  private final String mProfile;
+
+  public WiimoteProfileStringSetting(String gameID, int padID, String section, String key,
+          String defaultValue)
+  {
+    mPadID = padID;
+    mSection = section;
+    mKey = key;
+    mDefaultValue = defaultValue;
+
+    mProfileKey = SettingsFile.KEY_WIIMOTE_PROFILE + (padID + 1);
+
+    mProfile = gameID + "_Wii" + padID;
+  }
+
+  @Override
+  public boolean isOverridden(Settings settings)
+  {
+    return settings.isWiimoteProfileEnabled(settings, mProfile, mProfileKey);
+  }
+
+  @Override
+  public boolean isRuntimeEditable()
+  {
+    return false;
+  }
+
+  @Override
+  public boolean delete(Settings settings)
+  {
+    return settings.disableWiimoteProfile(settings, mProfileKey);
+  }
+
+  @Override
+  public String getString(Settings settings)
+  {
+    if (settings.isWiimoteProfileEnabled(settings, mProfile, mProfileKey))
+      return settings.getWiimoteProfile(mProfile, mPadID).getString(mSection, mKey, mDefaultValue);
+    else
+      return mDefaultValue;
+  }
+
+  @Override
+  public void setString(Settings settings, String newValue)
+  {
+    settings.getWiimoteProfile(mProfile, mPadID).setString(mSection, mKey, newValue);
+
+    settings.saveWiimoteProifleSettings();
```

Comment:
There isn't really a need to call this here, now that we're calling it in `Settings.saveSettings`. Calling it in `Settings.saveSettings` makes the most sense to me since it better matches how normal settings work.

Reply:
I thought the same thing too but removing that line of code prevents the extension setting from updating when backing out of an extension submenu. At least I'm pretty sure that was the problem, I'll verify when I cleanup this PR.

Reply:
Ah, right, that's because `getWiimoteProfile` is reloading from disk every time it's called. If you change that, you can remove this.